src/backend/joanie/core/views.py

```diff
@@ -45,77 +62,66 @@ class DebugMailSuccessPaymentViewTxt(DebugMailSuccessPayment):
     template_name = "mail/text/order_validated.txt"
 
 
-class DebugCertificateDefinitionTemplateView(TemplateView):
+class DebugTemplateView(TemplateView):
     """
-    Debug View to check the layout of certificate definition templates
+    Simple class to render the PDF template in bytes format of a document to preview.
+    Issuer document can be a : certificate, degree, invoice, or a contract definition.
     """
 
-    type = None
+    model = None
+    issuer_document = None
+    template_name = "debug/pdf_viewer.html"
 
-    # pylint: disable=invalid-name
-    def retrieve_certificate_from_pk(self, pk, site):
+    def get_document_context(self):
```

Comment:
```suggestion
    def get_document_context(self, pk=None):
```

According to the classes implementing it.